src/NuGet.Core/NuGet.Protocol/Resources/OwnerDetailsUriTemplateResourceV3.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Protocol.Resources
+{
+    /// <summary>Owner Details Uri Template for NuGet V3 HTTP feeds.</summary>
+    /// <remarks>Not intended to be created directly. Use <see cref="SourceRepository.GetResourceAsync{T}(CancellationToken)"/>
+    /// with <see cref="OwnerDetailsUriTemplateResourceV3"/> for T, and typecast to this class.
+    public class OwnerDetailsUriTemplateResourceV3 : INuGetResource
+    {
+        private readonly string _template;
+
+        private OwnerDetailsUriTemplateResourceV3(string template)
+        {
+            _template = template ?? throw new ArgumentNullException(nameof(template));
+        }
+
+        /// <summary>
+        /// Creates the specified Owner Details Uri template provided by the server if it exists and is valid.
+        /// </summary>
+        /// <param name="uriTemplate">The Absolute Uri template provided by the server.</param>
+        /// <returns>A valid Owner Details Uri template, or null.</returns>
+        public static OwnerDetailsUriTemplateResourceV3? CreateOrNull(Uri? uriTemplate)
```

Comment:
I like that this class implements nullable. 

I think the even more complete approach would be to just be null safe and throw Uri is null, and make the call ensure a Uri is passed when available. 

I know it may seem stylistic a bit, but the general idea is you want to minimize the places in your code that can work around null values.

Reply:
See https://github.com/NuGet/NuGet.Client/pull/5763#discussion_r1581639227